packages/components/src/typography/typography.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { ReactNode } from "react";
+
+import cx from "classnames";
+
+type TypographyElement = "h1" | "h2" | "h3" | "h4" | "h5" | "h6" | "span" | "p";
+
+type TypographySize =
+  | "xxxl"
+  | "xxl"
+  | "xl"
+  | "lg"
+  | "md"
+  | "sm"
+  | "xs"
+  | "xxs"
+  | "base";
+
+// Matches color token categories
+type TypographyColor =
+  | "alert"
+  | "success"
+  | "info"
+  | "warning"
+  | "brand"
+  | "black"
+  | "white";
+
+type TypographyProps = {
+  /**
+   * This prop can be used to specify which size text element should
+   * actually be rendered, in the case that you want to render an element
+   * as one heading but style it as if it were another. If both an `as` prop
+   * and a `size` prop are passed, the `as` will be used to determine the element
+   * and the `size` will be used to determine the classname.
+   */
+  as?: TypographyElement;
+  /**
+   * The text content to present.
+   */
+  children: ReactNode;
+  /**
+   * The color of the text element. If no color provided, a default color
+   * will be used.
+   */
+  color?: TypographyColor;
+  /**
+   * The size of the html element. If no `as` prop is provided, then
+   * the component uses this value to determine which html tag to render
+   * (e.g. 'h1', 'h2', etc.)
+   */
+  size: TypographySize;
+};
+
+function elementFromSize(size: TypographySize): TypographyElement {
+  switch (size) {
+    case "xxxl":
+      return "h1";
+    case "xxl":
+      return "h2";
+    case "xl":
+      return "h3";
+    case "lg":
+      return "h4";
+    case "md":
+      return "h5";
+    case "base":
+    case "sm":
+    case "xs":
+    case "xxs":
+      return "p";
+    default:
+      throw new TypeError(`Received invalid size option '${size}'`);
+  }
+}
+
+function Typography(props: TypographyProps): JSX.Element {
+  const {
+    as,
+    children,
+    color,
+    size,
+    /**
+     * Components that wrap typography sometimes requires props such as event handlers
+     * to be passed down into the element. One example is the tooltip component.  It
+     * attaches a onHover and onFocus event to the element to determine when to
+     * trigger the overlay.
+     */
+    ...rest
+  } = props;
+
+  const className = cx(`czedi-text-${size}`, color);
```

Comment:
I'm not sure the best way to handle color ... I didn't want to enumerate every option here as a potential class. But this current approach doesn't feel great

Reply:
this is where we need to [start applying BEM](http://getbem.com/naming/) `.czedi-text--${color}`. Enumerating them may be cumbersome, but it'll allow us to be explicit - and we could iterate over them with the right `scss` syntax.

Reply:
ahh ok will try

Reply:
updated! that definitely makes more sense